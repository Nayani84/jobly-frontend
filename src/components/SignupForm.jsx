import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import "./SignupForm.css";

function SignupForm({ signup }) {
  const navigate = useNavigate();
  const INITIAL_STATE = {
    username: '',
    password: '',
    firstName: '',
    lastName: '',
    email: ''
  }

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value
    }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    await signup(formData);
    navigate("/");
  };

  return (
    <div className="Signup-container">
      <h1>Sign Up</h1>
      <form onSubmit={handleSubmit}>

        <div>
          <label htmlFor="username">
            Username :
          </label>
          <input
            id="username"
            name="username"
            type="text"
            placeholder="Username"
            autoComplete="username"
            value={formData.username}
            onChange={handleChange}
            required
          />
        </div>

        <div>
          <label htmlFor="password">
            Password :
          </label>
          <input
            id="password"
            name="password"
            type="password"
            placeholder="Password"
            autoComplete="new-password"
            value={formData.password}
            onChange={handleChange}
            required
          />
        </div>

        <div>
          <label htmlFor="firstName">
            First Name :
          </label>
          <input
            id="firstName"
            name="firstName"
            type="text"
            placeholder="First Name"
            value={formData.firstName}
            onChange={handleChange}
            required
          />
        </div>

        <div>
          <label htmlFor="lastName">
            Last Name :
          </label>
          <input
            id="lastName"
            name="lastName"
            type="text"
            placeholder="Last Name"
            value={formData.lastName}
            onChange={handleChange}
            required
          />
        </div>

        <div>
          <label htmlFor="email">
            E-mail :
          </label>
          <input
            id="email"
            name="email"
            type="text"
            placeholder="email"
            autoComplete="email"
            value={formData.email}
            onChange={handleChange}
            required
          />
        </div>

        <button type="submit">Sign up</button>
      </form>
    </div>

  );
}

export default SignupForm;
