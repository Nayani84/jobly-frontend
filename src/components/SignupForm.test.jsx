import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SignupForm from "./SignupForm";
import { vi } from "vitest";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <SignupForm signup={() => { }} />
        </MemoryRouter>
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <SignupForm signup={() => { }} />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("runs the signup function on form submit", async function () {
    const signupMock = vi.fn();
    const { getByLabelText, getByText } = render(
        <MemoryRouter>
            <SignupForm signup={signupMock} />
        </MemoryRouter>
    );

    const usernameInput = getByLabelText("Username :");
    const passwordInput = getByLabelText("Password :");
    const firstNameInput = getByLabelText("First Name :");
    const lastNameInput = getByLabelText("Last Name :");
    const emailInput = getByLabelText("E-mail :");
    const submitButton = getByText("Sign up");

    fireEvent.change(usernameInput, { target: { value: "testuser" } });
    fireEvent.change(passwordInput, { target: { value: "password123" } });
    fireEvent.change(firstNameInput, { target: { value: "Test" } });
    fireEvent.change(lastNameInput, { target: { value: "User" } });
    fireEvent.change(emailInput, { target: { value: "testuser@example.com" } });

    fireEvent.click(submitButton);

    expect(signupMock).toHaveBeenCalledWith({
        username: "testuser",
        password: "password123",
        firstName: "Test",
        lastName: "User",
        email: "testuser@example.com",
    });
});
