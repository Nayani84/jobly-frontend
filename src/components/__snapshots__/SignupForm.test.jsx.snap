// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Signup-container"
  >
    <h1>
      Sign Up
    </h1>
    <form>
      <div>
        <label
          for="username"
        >
          Username :
        </label>
        <input
          autocomplete="username"
          id="username"
          name="username"
          placeholder="Username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="password"
        >
          Password :
        </label>
        <input
          autocomplete="new-password"
          id="password"
          name="password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
      </div>
      <div>
        <label
          for="firstName"
        >
          First Name :
        </label>
        <input
          id="firstName"
          name="firstName"
          placeholder="First Name"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="lastName"
        >
          Last Name :
        </label>
        <input
          id="lastName"
          name="lastName"
          placeholder="Last Name"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          E-mail :
        </label>
        <input
          autocomplete="email"
          id="email"
          name="email"
          placeholder="email"
          required=""
          type="text"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Sign up
      </button>
    </form>
  </div>
</DocumentFragment>
`;
